src/application/utils/transaction.ts

```diff
@@ -17,56 +17,127 @@ import {
   getSats,
   getAsset,
   NetworkString,
+  IdentityInterface,
 } from 'ldk';
 import { confidential, networks, payments, Psbt } from 'liquidjs-lib';
-import { blindingKeyFromAddress, isConfidentialAddress } from './address';
+import { isConfidentialAddress } from './address';
 import { Transfer, TxDisplayInterface, TxStatusEnum, TxType } from '../../domain/transaction';
 import { Topup } from 'taxi-protobuf/generated/js/taxi_pb';
 import { lbtcAssetByNetwork } from './network';
-import { fetchTopupFromTaxi } from './taxi';
-import { taxiURL } from './constants';
+import { fetchTopupFromTaxi, taxiURL } from './taxi';
 import { DataRecipient, isAddressRecipient, isDataRecipient, Recipient } from 'marina-provider';
 
-function outPubKeysMap(pset: string, outputAddresses: string[]): Map<number, string> {
-  const outPubkeys: Map<number, string> = new Map();
+const blindingKeyFromAddress = (addr: string): Buffer => {
+  return address.fromConfidential(addr).blindingKey;
+};
+
+function outPubKeysMap(pset: string, outputAddresses: string[]): Map<number, Buffer> {
+  const outPubkeys: Map<number, Buffer> = new Map();
 
-  for (const outAddr of outputAddresses) {
-    const index = outputIndexFromAddress(pset, outAddr);
+  for (const outAddress of outputAddresses) {
+    const index = outputIndexFromAddress(pset, outAddress);
     if (index === -1) continue;
-    if (isConfidentialAddress(outAddr)) {
-      const blindingPublicKey = blindingKeyFromAddress(outAddr);
-      outPubkeys.set(index, blindingPublicKey);
+    if (isConfidentialAddress(outAddress)) {
+      outPubkeys.set(index, blindingKeyFromAddress(outAddress));
     }
   }
 
   return outPubkeys;
 }
 
+function inputBlindingDataMap(
+  pset: string,
+  utxos: UnblindedOutput[]
+): Map<number, confidential.UnblindOutputResult> {
+  const inputBlindingData = new Map<number, confidential.UnblindOutputResult>();
+  const txidToBuffer = function (txid: string) {
+    return Buffer.from(txid, 'hex').reverse();
+  };
+
+  let index = -1;
+  for (const input of psetToUnsignedTx(pset).ins) {
+    index++;
+    const utxo = utxos.find((u) => txidToBuffer(u.txid).equals(input.hash));
+    if (!utxo) {
+      throw new Error(`blindPSET error: utxo not found '${input.hash.reverse().toString('hex')}'`);
+    }
```

Comment:
There may be occasions where the blinding is either a) not done by us (ccoperative blinding) b) not blinded (like taxi transactions for example)

for a) it's fine to wait new PSETv2 specification that cleanly allows to do this, but for b) we must skip throwing error for unblinded inputs we do not control (ie. provided by taxi)

@louisinger

Reply:
Just a quick note: this code makes impossible to use Taxi, since first input from psetToUnsignedTx is from a utxo the wallet doesn't have (it comes from Taxi), so it will always throw in the case of a Taxi transaction.

I remove it and tested all type of transactions, and everything is working.

Just not sure if by removing it I'm forgetting any edge case. @louisinger ?

Reply:
Indeed throwing an error is not OK in case of unconfidential inputs (taxi especially). However, in case of **confidential input we don't own**, blinding will fail if we are not able to give it the blinders (utxo.blindingData). 
Now, it's not a problem cause we always use Marina's inputs except the taxi ones but they are unconfidential. if in the future marina integrates a feature that uses confidential inputs from another wallet, it will be necessary to modify this function (add a parameter "externalInputsBlindingPrivateKeys" for instance). But I think it's better to wait for PSETv2 for this case as suggested by tiero (which would allow not to manage the blinding keys / blinders exchange).

TL;DR @bordalix no edge cases because Marina never handle cases of confidential inputs owned by another wallet. ACK to remove the check for me :+1: